pgt/src/PGT/Output/TestSuite.hs

```diff
@@ -25,22 +26,28 @@ import qualified Test.Tasty            as Tasty
 
 data TestSuite a = TestSuite
   { definitions :: [Text]
-  , tests       :: NonEmpty (Test a)
+  , tests       :: [Test a]
   }
   deriving stock (Eq, Show)
 
 instance Render (TestSuite QueryStats) where
-  render TestSuite{..} = unlinesN 2 $ definitions <> [renderTests]
+  render TestSuite{..} = Text.stripEnd . unlinesN 2 $ definitions <> [renderTests]
     where
       renderTests :: Text
       renderTests = unlinesN 3 . Foldable.toList $ render <$> tests
 
 parse :: Parser (TestSuite QueryStats)
 parse = do
-  definitions <- Text.many' (Definition.parse <* Text.endOfLine)
+  definitions <- Text.many' (Definition.parse <* (Text.endOfInput <|> Text.endOfLine))
+  atEnd       <- Text.atEnd
 
-  either fail (pure . TestSuite definitions . NonEmpty.fromList)
-    =<< Text.eitherP (parseUnexpected (listToMaybe definitions)) parseTests
+  if | atEnd && Foldable.null definitions ->
+        fail "expected a valid test or database object definition but found none"
+     | atEnd                              ->
+        pure $ TestSuite definitions []
+     | otherwise                          ->
+        either fail (pure . TestSuite definitions)
+          =<< Text.eitherP (parseUnexpected (listToMaybe definitions)) parseTests
```

Comment:
Comment: The `multi-way if` enables us to be more string in the parsing of expected present `TestSuite.tests`